recipes/default.rb

```diff
@@ -15,3 +15,174 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+#
+
+include_recipe 'osl-apache'
+include_recipe 'osl-apache::mod_ssl'
+include_recipe 'osl-apache::mod_perl'
+include_recipe 'osl-mysql::client'
+include_recipe 'yum-osuosl'
+include_recipe 'perl'
+
+package %w(request-tracker mutt procmail)
+
+# Get the config options, and overwrite the options given with other properties
+config_options = {}
+
+# Organization will be the Main domain name only, no sub-level domains.
+
+config_options['$rtname'] = node['osl-rt']['fqdn']
+config_options['$WebDomain'] = node['osl-rt']['fqdn']
+config_options['$Organization'] = node['osl-rt']['fqdn'][/([\w\-_]+\.+\w+$)/]
+config_options['$CorrespondAddress'] = "#{node['osl-rt']['default']}@#{config_options['$Organization']}"
+config_options['$CommentAddress'] = "#{node['osl-rt']['default']}-comment@#{config_options['$Organization']}"
+config_options['$DatabaseType'] = node['osl-rt']['db']['type']
+config_options['$DatabaseHost'] = node['osl-rt']['db']['host']
+config_options['$DatabaseRTHost'] = node['osl-rt']['db']['host']
+config_options['$DatabaseName'] = node['osl-rt']['db']['name']
+config_options['$DatabaseUser'] = node['osl-rt']['db']['username']
+config_options['$DatabasePassword'] = node['osl-rt']['db']['password']
+config_options['_Plugins'] = node['osl-rt']['plugins']
+config_options['_Lifecycles'] = node['osl-rt']['lifecycles']
+
+# Set up the queue emails
+rt_emails = init_emails(node['osl-rt']['queues'], node['osl-rt']['fqdn'], node['osl-rt']['default'])
+
+config_options['$RTAddressRegexp'] = "^(#{rt_emails.join('|')}(-comment)?\@(#{node['osl-rt']['fqdn']}))"
+
+# Set up the hostname as the website
+hostname node['hostname'] do
+  aliases ["#{node['osl-rt']['fqdn']}"]
+end
+
+# Parse the config hash to a string we can use as
+# the RT_SiteConfig.pm file.
+strConfigFile = parse_config(config_options)
+
+# Root Account
+template '/root/.rtrc' do
+  source 'rt/rtrc.erb'
+  cookbook 'osl-rt'
+  user 'root'
+  group 'root'
+  mode '0600'
+  sensitive true
+  variables(root_pass: node['osl-rt']['root-password'], domain: node['osl-rt']['fqdn'])
+end
+
+# Add the RT command to the root user's PATH
+link '/usr/local/sbin/rt' do
+  to '/opt/rt/bin/rt'
+end
+
+# RT Initial Configuration.
+file '/opt/rt/etc/RT_SiteConfig.pm' do
+  content strConfigFile
+  user 'root'
+  group 'apache'
+  mode '0640'
+  sensitive true
+  notifies :reload, 'apache2_service[osuosl]'
+end
+
+# Initalize the DB
+execute 'init-db-rt' do
+  command <<-EOH
```

Comment:
This properly removes the prepended spaces when executing. I usually like doing this if possible as it makes reading the output (later if needed) easier.
```suggestion
  command <<~EOH
```